File: converter-dotnet-core-console-cs/ConvertAsyncExample.cs
using ceTe.DynamicPDF.Conversion;
using System;
using System.IO;
using System.Threading.Tasks;

namespace converter_dotnet_core_console_cs
{
    public class ConvertAsyncExample
    {
        public static void Run()
        {
          
           var tsk1 = RunTextSaveFile(@"async-text-file-output.pdf");
           var tsk2 = RunTextInputFileName(@"async-text-input-file-name-output.pdf");
           var tsk3 = RunTextBytesConvOpt(@"async-text-bytes-options-output.pdf");
           var tsk4 = RunTextBytesConvAppnd(@"async-text-bytes-append-output.pdf");
           Task.WaitAll(new Task[] {tsk2, tsk1, tsk3, tsk4});
           Console.WriteLine("Conversion Completed successfully");
        }

        public static async Task RunTextInputFileName(string outputPdf)
        {
            byte[] input = File.ReadAllBytes(Program.GetResourcePath(@"DocumentA.txt"));
            byte[] result = await Converter.ConvertAsync(input,Program.GetResourcePath(@"DocumentA.txt"));

            if (result != null)
            {
               File.WriteAllBytes(Program.GetOutputDocPath(outputPdf), result);
            }

        }

        public static async Task RunTextBytesConvOpt(string outputPdf)
        {
            ConversionOptions options = new(false);
            byte[] result = await Converter.ConvertAsync(Program.GetResourcePath(@"DocumentA.txt"), options);

            if (result != null)
            {
                File.WriteAllBytes(Program.GetOutputDocPath(outputPdf), result);

            }
        }


        public static async Task RunTextBytesConvAppnd(string outputPdf)
        {
            ConversionOptions options = new(false);
            byte[] appendToPdf = File.ReadAllBytes(Program.GetResourcePath(@"DocumentA.pdf"));

            byte[] result = await Converter.ConvertAsync(Program.GetResourcePath(@"DocumentA.txt"), options, appendToPdf);

            if (result != null)
            {
                File.WriteAllBytes(Program.GetOutputDocPath(outputPdf), result);

            }
        }


        public static async Task RunTextSaveFile(string outputPdf)
        {

            bool result = await Converter.ConvertAsync(Program.GetResourcePath(@"DocumentA.rtf"), Program.GetOutputDocPath(outputPdf));

            if (result)
            {
                Console.WriteLine("Conversion Completed successfully");
            }
        }

    }
}

File: converter-dotnet-core-console-vb/ConvertAsyncExample.vb
Imports ceTe.DynamicPDF.Conversion
Imports System
Imports System.IO

Public Class ConvertAsyncExample
    Public Shared Sub Run()
        Dim tsk1 As Task = RunTextSaveFile("async-text-file-output.pdf")
        Dim tsk2 As Task = RunTextInputFileName("async-text-input-file-name-output.pdf")
        Dim tsk3 As Task = RunTextBytesConvOpt("async-text-bytes-options-output.pdf")
        Dim tsk4 As Task = RunTextBytesConvAppnd("async-text-bytes-append-output.pdf")

        Task.WaitAll({tsk2, tsk1, tsk3, tsk4})
        Console.WriteLine("Conversion Completed successfully")
    End Sub

    Public Shared Async Function RunTextInputFileName(outputPdf As String) As Task
        Dim input As Byte() = File.ReadAllBytes(Program.GetResourcePath("DocumentA.txt"))
        Dim result As Byte() = Await Converter.ConvertAsync(input, Program.GetResourcePath("DocumentA.txt"))

        If result IsNot Nothing Then
            File.WriteAllBytes(Program.GetOutputDocPath(outputPdf), result)
        End If
    End Function

    Public Shared Async Function RunTextBytesConvOpt(outputPdf As String) As Task
        Dim options As New ConversionOptions(False)
        Dim result As Byte() = Await Converter.ConvertAsync(Program.GetResourcePath("DocumentA.txt"), options)

        If result IsNot Nothing Then
            File.WriteAllBytes(Program.GetOutputDocPath(outputPdf), result)
        End If
    End Function

    Public Shared Async Function RunTextBytesConvAppnd(outputPdf As String) As Task
        Dim options As New ConversionOptions(False)
        Dim appendToPdf As Byte() = File.ReadAllBytes(Program.GetResourcePath("DocumentA.pdf"))

        Dim result As Byte() = Await Converter.ConvertAsync(Program.GetResourcePath("DocumentA.txt"), options, appendToPdf)

        If result IsNot Nothing Then
            File.WriteAllBytes(Program.GetOutputDocPath(outputPdf), result)
        End If
    End Function

    Public Shared Async Function RunTextSaveFile(outputPdf As String) As Task
        Dim result As Boolean = Await Converter.ConvertAsync(Program.GetResourcePath("DocumentA.rtf"), Program.GetOutputDocPath(outputPdf))

        If result Then
            Console.WriteLine("Conversion Completed successfully")
        End If
    End Function
End Class
